Lib/fontMath/mathGlyph.py

```diff
@@ -388,18 +397,21 @@ def _flushContour(self):
         holdingOffCurves = []
         for index, point in enumerate(points):
             segmentType = point[0]
-            if segmentType == "line":
-                pt, smooth, name, identifier = point[1:]
-                prevPt = points[index - 1][1]
-                if index == 0:
-                    holdingOffCurves.append((None, prevPt, False, None, None))
-                    holdingOffCurves.append((None, pt, False, None, None))
-                else:
-                    contourPoints.append((None, prevPt, False, None, None))
-                    contourPoints.append((None, pt, False, None, None))
-                contourPoints.append(("curve", pt, smooth, name, identifier))
-            else:
+            if self.strict:
                 contourPoints.append(point)
+            else:
+                if segmentType == "line":
```

Comment:
you could do in a single `if` condition, since the `else` condition is the same (append the point as is) whether its a non-line or it's strict mode; e.g.

```python
if segmentType == "line" and not self.strict:
    ...
else:
    contourPoints.append(point)
```

Reply:
Yes, that's nicer. Changed.